pixi/backend/api.js

```diff
@@ -15,17 +15,78 @@
  */
 
 const express = require('express');
+const {lint, LintMode, StatusNumber} = require('@ampproject/toolbox-linter');
+const cheerio = require('cheerio');
 const log = require('@lib/utils/log')('Pixi API');
+const LimitedRemoteFetch = require('@lib/utils/limitedRemoteFetch');
+
+const remoteFetch = new LimitedRemoteFetch({
+  requestHeaders: {
+    'Referer': 'https://amp.dev/page-experience/',
+  },
+});
+
+const execLint = async (url) => {
+  const res = await remoteFetch.fetchHtmlResponse(url);
+  const body = await res.text();
+  const context = {
+    $: cheerio.load(body),
+    headers: {},
+    raw: {
+      headers: res.headers,
+      body,
+    },
+    url,
+    mode: LintMode.Amp,
+  };
+  return lint(context);
+};
+
+const mapLintResult = (result) => {
+  return Object.entries(result).reduce((mappedData, [key, checks]) => {
+    if (Array.isArray(checks)) {
+      mappedData[key] = checks
+        .map((item) => item.status)
+        .reduce((result, status) => {
+          if (!status) {
+            return result;
+          }
+          if (!result || StatusNumber[result] < StatusNumber[status]) {
+            return status;
+          }
+          return result;
+        });
+    } else {
+      mappedData[key] = checks.status;
+    }
+    return mappedData;
+  }, {});
+};
 
 // eslint-disable-next-line new-cap
 const api = express.Router();
-api.get('/hello-world', async (request, response) => {
-  log.info('hello-world endpoint called.');
-
+api.get('/lint', async (request, response) => {
+  log.info('lint endpoint called.');
   response.setHeader('Content-Type', 'application/json');
-  response
-    .status(200)
-    .send(JSON.stringify({'message': 'Hello World'}, null, 2));
+
+  const fetchUrl = request.query.url;
+  try {
+    const lintResult = await execLint(fetchUrl);
+    const result = {
+      status: 'ok',
+      data: mapLintResult(lintResult),
+    };
+    response.status(200).send(JSON.stringify(result, null, 2));
```

Comment:
Use `response.json(result)` this will automatically set status 200 and the correct content type.